File: languages/go/entity/identityV2/schema.go
package identityV2

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

type Schema struct {
	Cid        string
	CidJsonLd  string
	SchemaType string
	Json       string
}

func NewSchemaFromProto(s *proto.SchemaV2) Schema {
	if s == nil {
		return Schema{}
	}
	return Schema{
		Cid:        s.Cid,
		CidJsonLd:  s.CidJsonLd,
		SchemaType: s.SchemaType,
		Json:       s.Json,
	}
}

func (c Schema) ToProto() *proto.SchemaV2 {
	return &proto.SchemaV2{
		Cid:        c.Cid,
		CidJsonLd:  c.CidJsonLd,
		SchemaType: c.SchemaType,
		Json:       c.Json,
	}
}

File: languages/python/bloock/entity/identity_v2/schema.py
from __future__ import annotations

import bloock._bridge.proto.identity_entities_v2_pb2 as proto


class Schema:
    def __init__(self, cid: str, cid_json_ld: str, json: str) -> None:
        self.cid = cid
        self.cid_json_ld = cid_json_ld
        self.json = json

    @staticmethod
    def from_proto(c: proto.SchemaV2) -> Schema:
        return Schema(
            cid=c.cid,
            cid_json_ld=c.cid_json_ld,
            json=c.json,
        )

    def to_proto(self) -> proto.SchemaV2:
        return proto.SchemaV2(
            cid=self.cid,
            cid_json_ld=self.cid_json_ld,
            json=self.json,
        )
